reference-implementation/lib/transform-stream.js

```diff
@@ -16,6 +17,17 @@ function TransformStreamCloseReadable(transformStream) {
     throw new TypeError('Readable side is already closed');
   }
 
+  if (transformStream._writableDone === true) {
+    throw new TypeError('TransformStream is already closing');
+  }
```

Comment:
Having two redundant ways to perform the same action seemed dubious (especially if, as in your other comment, the result is potentially contradictory, instead of redundant). It's already an error to call `close()` twice, this seemed roughly analogous to that.